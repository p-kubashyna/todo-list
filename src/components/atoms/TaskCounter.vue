<script setup>
import { useStore } from "vuex";
import { computed } from "vue";

const store = useStore();
const incompleteCounter = computed(() => store.getters.getIncompleteTodosCount);

function getLabel() {
  if (incompleteCounter.value === 0) {
    return "No tasks left";
  } else if (incompleteCounter.value === 1) {
    return "1 task left";
  } else {
    return incompleteCounter.value + " tasks left";
  }
}
</script>

<template>
  <span class="counter-label">{{ getLabel() }}</span>
</template>

<style scoped>
.counter-label {
  font-family: monospace;
  color: #121952;
  margin-left: 10px;
  margin-right: auto;
}
</style>
