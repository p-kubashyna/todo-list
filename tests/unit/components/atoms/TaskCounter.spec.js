import { describe, expect, it } from "@jest/globals";
import { mount } from "@vue/test-utils";
import { createStore } from "vuex";
import TaskCounter from "@/components/atoms/TaskCounter.vue";

describe("Given the TaskCounter component", () => {
  it("should render", async () => {
    const store = createStore({
      getters: {
        getIncompleteTodosCount: () => 0,
      },
    });

    const wrapper = mount(TaskCounter, {
      global: {
        plugins: [store],
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it.each([
    [0, "No tasks left"],
    [1, "1 task left"],
    [2, "2 tasks left"],
  ])("should have correct label", async (counterValue, labelText) => {
    const store = createStore({
      getters: {
        getIncompleteTodosCount: () => counterValue,
      },
    });

    const wrapper = mount(TaskCounter, {
      global: {
        plugins: [store],
      },
    });

    expect(wrapper.find("span").text()).toBe(labelText);
  });
});
