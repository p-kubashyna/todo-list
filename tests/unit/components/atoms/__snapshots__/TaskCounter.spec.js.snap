// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the TaskCounter component should render 1`] = `
<span
  class="counter-label"
>
  No tasks left
</span>
`;
